external_libraries/CMakeLists.txt

```diff
@@ -104,7 +105,11 @@ if(NOT YAMLCPP_FOUND)
   CREATE_FINAL_FILE(${CMAKE_CURRENT_BINARY_DIR}/libyamlcpp.cpp ${yaml_src})
 
   add_library(yaml STATIC EXCLUDE_FROM_ALL ${CMAKE_CURRENT_BINARY_DIR}/libyamlcpp.cpp)
-  target_include_directories(yaml PUBLIC ${CMAKE_SOURCE_DIR}/external_libraries/yaml-cpp/include boost)
+  if(SYSTEM_BOOST)
+    target_include_directories(yaml PUBLIC ${Boost_INCLUDE_DIRS})
+  else()
+    target_include_directories(yaml PUBLIC ${CMAKE_SOURCE_DIR}/external_libraries/yaml-cpp/include boost)
+  endif()
```

Comment:
in both cases, you need the `external_libraries` include, otherwise building fails if you:
- have yaml-cpp in a reachable directory in an older version
- use SYSTEM_BOOST
- don't use SYSTEM_YAMLCPP

this happened to me :)

Reply:
Wouldn't this also mean unconditionally including the local version of yaml-cpp?

Reply:
the message at the top of this block says "using bundled libyaml-cpp", so i'm assuming that is intended here.